docs/blog/20240624-Daeraxa-3kStars.md

```diff
@@ -0,0 +1,136 @@
+
+Of course, there was the main technical challenge, but this wasn't the only obstacle; we had other things that had to be taken into account, like what to do about packages with [no license](https://pulsar-edit.dev/blog/20230319-confused-Techie-HowLicenseNoneDeletedPackages.html), or custom licenses specifically for Atom.io? We also had to deal with a huge amount of spam from Vietnamese and Thai gambling sites that had been hammering the original backend in its final days with no moderation. To indicate the scale of the issue, 12k packages were archived compared to 405k just before atom.io finally shut down.
+
+### Modernizing the editor
+
+<img src="./assets/pulsar1.png" height="200" />
+
+We had some big tasks ahead of us; changes needed to be made, and work needed to start to get some traction on the project. We desperately wanted to keep the community alive, but with the announcement of the sunset, people were fleeing the sinking ship. We had to get the project up and running to show that Pulsar was a serious continuation of the project and not just a fork on GitHub with a renamed readme.
+
+- **Building from source** - Atom had a rather complicated (and now rather fragile) way of [bootstrapping](https://atom-flight-manual-archive.github.io/hacking-atom/sections/hacking-on-atom-core/#cloning-and-bootstrapping) itself for development. One of the first tasks was to simplify this into a process much more familiar to those using Node.js and Electron, which is far more robust and makes it much easier to get involved with Pulsar development.
+- **Rebranding** - We had to start removing Atom references and replacing them with Pulsar ones. This was not an easy "find and replace" job as `atom` is used extensively within the APIs and code and therefore something we had to be careful of. We also had to decide on a new logo for the project and add it to the application, along with updates to redirect links within Pulsar to our new repositories, websites, forums, etc.
+- **Decaffeination** - Atom was originally written mostly in [CoffeeScript](https://coffeescript.org/), a rather useful language at the time that compiles to JavaScript. With more and more functionality being introduced into JavaScript, CoffeeScript was quickly becoming obsolete and actually started creating a barrier to entry for contribution rather than easing it, as was the initial intention. The original Atom team had started this process, but over time we were able to achieve the goal of complete decaffeination in Pulsar 1.109 and in PPM in 1.110.
+- **CI and build** - We needed an automated platform to build our binaries and run tests. For this, we went with [CirrusCI](https://cirrus-ci.org/) which had a more than generous free tier, and it reliably built our binaries and ran our tests for over a year. We have since had to [move away](https://pulsar-edit.dev/blog/20230903-confused-Techie-pulsars-ci.html) from CirrusCI.
```

Comment:
```suggestion
- **CI and builds:** - We needed an automated platform to build our binaries and run tests. For this, we went with [CirrusCI](https://cirrus-ci.org/), which had a more than generous free tier; it reliably built our binaries and ran our tests for over a year. We have since had to [move away](https://pulsar-edit.dev/blog/20230903-confused-Techie-pulsars-ci.html) from CirrusCI for most tasks, but we still use it for building some of our binaries.

```

Reply:
I meant to add that in that we were still using it for some things... glad you caught it